src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Directives.cs

```diff
@@ -285,7 +289,7 @@ protected void BaseTypeDirective(string noTypeNameError, Func<string, SpanChunkG
             Output(SpanKind.Code, AcceptedCharacters.AnyExceptNewline);
         }
 
-        private void TagHelperDirective(string keyword, ISpanChunkGenerator chunkGenerator)
+        private void TagHelperDirective(string keyword, Func<string, ISpanChunkGenerator> chunkGeneratorThunk)
```

Comment:
Such an awesome name. I can change it to `chunkGeneratorFactory` if that's the more common term in the codebase.